store.go

```diff
@@ -38,6 +40,9 @@ type DefaultConfigStore struct {
 
 	// aliases contains the collection of any aliased config values
 	aliases map[string]string
+
+	// if a store which is logable is desired
+	log *DefaultLogger
```

Comment:
I don't think that these logging specific bits really belong in the `DefaultConfigStore`. You should only need to wrap a `ConfigStore` with a `LogableConfigStore` that will handle the logging